### DISCUSSION TITLE
Towards v1.0.0 — Where are we? Where are we going?

### DISCUSSION THREAD

**Discussion by josefarias** [josefarias/hotwire_combobox#200] [Announcements]
Wanted to give everyone a sense of where we're at in building towards `v1.0.0`.

## What are we building?
The vision for this library is building an accessible autocomplete that's easy to use and will provide a single sanctioned way to do everything you'd reasonably want to do with a combobox in a RoR + Hotwire app. In my mind, that's a finite set of features. So I'm hoping at some point this'll be feature-complete.

The combobox should include sensible jumping-off points for customizing how the combobox looks, and for hooking into its behavior. But changing how the components are laid out/assembled, or significantly altering the behavior of the comobox is not a priority. It's not meant to be an engine. But, rather, a powerful standalone component (TBD whether that includes a [headless version](https://github.com/josefarias/hotwire_combobox/discussions/113) of the component, see below).

## Where are we?
Sync search, async search, multiselect (including liveregions for better a11y), css customization, custom markup in options, custom attributes in every element, large touch targets on mobile, custom DOM events, freetext options, keyboard navigation, and HTML validations are all already supported. But not all of these are finalized.

Specifically, I've held off on documenting (and thus finalizing) multiselect, custom DOM events, and custom attributes just so we could live with them for a while and make any breaking decisions before people adopt them.

I think custom DOM events and custom attributes are stable enough. But with multiselect, I'd like to find a way to avoid making a roundtrip to the server for rendering vanilla chips (with no customizations). Lots of ways to go about this. But it's important to me we land on a simple implementation.

There are also some bugs I'd like to fix.

### Checklist for `v1.0.0`:
This is everything I'd like to happen before we release `v1.0.0`. But we'll likely have some minor version releases rolling these out as we go.

- [ ] Avoid making roundtrip to server for vanilla multiselect chips
- [ ] Document multiselect
- [ ] Document custom DOM events
- [ ] Document custom attributes
- [ ] Document validations
- [ ] [Deprecate enum cleverness](https://github.com/josefarias/hotwire_combobox/pull/160) (breaking change)
- [ ] Remove warning about the API changing and bugs being expected
- [x] [Update instructions so imported stimulus controller is not overwritten when updating manifest via CLI](https://github.com/josefarias/hotwire_combobox/pull/209)
- [ ] Fix https://github.com/josefarias/hotwire_combobox/discussions/167
- [x] Fix https://github.com/josefarias/hotwire_combobox/discussions/186
- [ ] Fix https://github.com/josefarias/hotwire_combobox/discussions/198
- [x] Fix https://github.com/josefarias/hotwire_combobox/discussions/199
- [ ] Fix https://github.com/josefarias/hotwire_combobox/issues/155
- [x] Merge https://github.com/josefarias/hotwire_combobox/issues/196
- [x] ~~Merge https://github.com/josefarias/hotwire_combobox/pull/179~~
- [x] Merge https://github.com/josefarias/hotwire_combobox/pull/235
- [x] Merge https://github.com/josefarias/hotwire_combobox/pull/209
- [ ] Cut `v1.0.0` release

### Small bits for `v1.x`:
- [Disabled options](https://github.com/josefarias/hotwire_combobox/discussions/126)
- [Better PORO support](https://github.com/josefarias/hotwire_combobox/discussions/153#discussioncomment-9216519)
- Consider [morphing in a different input value](https://github.com/josefarias/hotwire_combobox/issues/184#issuecomment-2267614129)
- [Reducing number of partials](https://github.com/josefarias/hotwire_combobox/discussions/194)
- [Fixing Android back button](https://github.com/josefarias/hotwire_combobox/issues/39)
- Dropping `requestjs` and `bodyScrollLock` as dependencies
- Sorting out some [bootstrap compatibility issues](https://github.com/josefarias/hotwire_combobox/discussions/84)

### Might be major-version worthy (`v2.x`):
- [Headless mode](https://github.com/josefarias/hotwire_combobox/discussions/113)

Thanks for being an active participant and reading this. You, specifically, are welcome to contribute however you'd like to any of these points. PRs are welcome, and so are bug reports in the [issues section](https://github.com/josefarias/hotwire_combobox/issues) and general opinions in our [discussion board](https://github.com/josefarias/hotwire_combobox/discussions).

**Comment by laptopmutia** [comment]
I wish it have a simpler css to override, and default css config like what pagy did with it extras
